inc/saiswitch.h

```diff
@@ -3048,6 +3048,33 @@ typedef enum _sai_switch_attr_t
      */
     SAI_SWITCH_ATTR_MAX_ICMP_ECHO_SESSION,
 
+    /**
+     * @brief Maximum size of the packet after trimming
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 128
+     */
+    SAI_SWITCH_ATTR_PACKET_TRIMMING_SIZE,
```

Comment:
Comments are confusing. One can interpret them as that trimming can be done lesser then configured byte upto maximum size.

We should change it keep it consistent with sample truncation attribute comments
"@brief Truncate size. Truncate sampled packets to this size to reduce traffic bandwidth"

Something like "Trim packet to this size"

Reply:
Also, should we add an attribute to specify a min size threshold for packets to qualify for trimming? As an example - a packet can only be trimmed if its size exceeds a min threshold Len1 and if trimmed, it will be trimmed to size Len2, where Len1 >= Len2.

Reply:
> Also, should we add an attribute to specify a min size threshold for packets to qualify for trimming? As an example - a packet can only be trimmed if its size exceeds a min threshold Len1 and if trimmed, it will be trimmed to size Len2, where Len1 >= Len2.

Ashutosh,

I am not sure about adding a min size. Truncation size is the min and max both. We use similar semantics for Sampling as well.

Reply:
Changed description